server/channels/app/user.go

```diff
@@ -2817,3 +2818,53 @@ func (a *App) UserIsFirstAdmin(user *model.User) bool {
 
 	return true
 }
+
+func (a *App) GetUsersForReporting(
+	sortColumn string,
+	sortDesc bool,
+	pageSize int,
+	lastSortColumnValue string,
+	lastUserId string,
+	startAt int64,
+	endAt int64,
+) ([]*model.UserReport, *model.AppError) {
+	// Don't allow fetching more than 100 users at a time from the normal query endpoint
+	if pageSize <= 0 || pageSize > 100 {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_users_for_reporting.invalid_page_size", nil, "", http.StatusBadRequest)
+	}
+
+	// Validate date range
+	if endAt > 0 && startAt > endAt {
+		return nil, model.NewAppError("GetUsersForReporting", "app.user.get_users_for_reporting.bad_date_range", nil, "", http.StatusBadRequest)
+	}
```

Comment:
Wondering if we check these at api level. Was there a specific reason implementing in the app layer?

Reply:
I would expect that anything calling this method should be limited to the same constraints, both for performance and to keep consistent with the API.

There will be some follow-up work to do some paged reading of this data into a CSV and I'd want to enforce the same constraints.

Reply:
This has been an inconsistent thing in our codebase where sometimes these checks are written in the api layer, and sometimes in the app layer, and even worse sometimes it's in both and we don't want to change it out of fear.

Overall I am 0/5, but would like to see consistency enforced via unit tests to get out of the fear of breaking things.